### PR TITLE
Gather image data in postgresql

### PR DESCRIPTION
**Pull request by mloviska**
Moving away from InfluxDB required several code changes in the image data gathering module.
Currently we use postgresql together with postgrest API in order to retrieve or push data from the test runs.

- ticket: https://progress.opensuse.org/issues/124595
- VR: http://kepler.suse.cz/tests/20434#

### CODE REVIEW

**Comment by mloviska** [comment]
VRs:
https://openqa.suse.de/tests/10642407#step/image_info/1
https://openqa.suse.de/tests/10642424#step/image_info/1

File: lib/db_utils.pm [hunk 1]
```diff
@@ -153,20 +155,22 @@ sub push_image_data_to_db {
...
-    $cmd .= '"type": "' . $args{type} . '", ';
-    $cmd .= '"value": "' . $value . '"}\'';
+    use Data::Dumper;
+    print("TESTING ...." . Dumper(%args));
+    assert_script_run('cat /etc/postgres_conf');
+    my $cmd = sprintf('curl -i -K /etc/postgres_conf -X POST http://%s:%s/%s', $db_ip, $db_port, $args{table});
+    delete $args{table};
+    $cmd .= " -d '";
+    $cmd .= encode_json(\%args);
+    $cmd .= q(');
```
Comment (ilausuch) [inline lib/db_utils.pm:172]: Maybe now is more readable put that in one single line
`$cmd .= " -d '" . encode_json(\%args) . q(');`

**Review by ilausuch — APPROVED** [review]
Good job!

File: lib/db_utils.pm (hunk 1 above)
Comment (mloviska) [reply lib/db_utils.pm:172]: oh, I have forgotten to remove the `Data::Dumper` debug lines xD
Yeah, well I have kinda kept the same logic that one line == one action. :) 
However, I do not mind to write it as you suggest :)

File: lib/db_utils.pm [hunk 2]
```diff
@@ -179,3 +181,51 @@ sub push_image_data_to_db {
...
+    my $query_output = script_output("$query 2>&1 | tee -a $db_log", proceed_on_failure => 1);
+
+    # Successful query should return 'HTTP/1.1 200 OK'
+    # Empty records will return  '{"results":[{"statement_id":0}]}'
+    # Existing records will return '{"results":[{"statement_id":0,"series":[{"name":"size","columns":["time","value"],"values"' ...
+    if ($query_output !~ /200 OK/) {
+        record_soft_failure("poo#110221 - There has been a problem with the query.");
+        return 0;
+    } elsif ($query_output =~ /values/) {
+        record_info('DB skip', "Skip pushing the data to the DB. There is already a DB record for this image.");
```
Comment (jlausuch) [inline lib/db_utils.pm:226]: Shouldn't this SKIP be part of the push method? We are not pushing anything here, right?

File: lib/db_utils.pm [hunk 3]
```diff
@@ -179,3 +181,51 @@ sub push_image_data_to_db {
...
+    elsif ($openqa_host =~ /openqa.suse.de/) {    # OSD hostname
+        $job_url = 'https://openqa.suse.de/tests/' . get_current_job_id();
+    }
+    return 0 unless ($job_url);
+
+    # Check if the data for this image has been already published before.
+    # This will avoid publishing the same data if a job is restarted.
+    # e.g.
+    # curl http://k2.qe.suse.de:8080/size\?product\=fts.jeos\&asset\=fts.SLES15-SP5-Minimal-VM.x86_64-kvm-and-xen-Build3.32.qcow2\&value\=eq.258.1875
+    my $query = sprintf('curl -I "http://%s:%s/size?product=fts.jeos&asset=fts.%s"', $db_ip, $db_port, $image);
```
Comment (jlausuch) [inline lib/db_utils.pm:214]: hardcoded `jeos`

File: lib/db_utils.pm [hunk 4]
```diff
@@ -179,3 +181,51 @@ sub push_image_data_to_db {
...
+    }
+    elsif ($openqa_host =~ /openqa.suse.de/) {    # OSD hostname
+        $job_url = 'https://openqa.suse.de/tests/' . get_current_job_id();
+    }
+    return 0 unless ($job_url);
+
+    # Check if the data for this image has been already published before.
+    # This will avoid publishing the same data if a job is restarted.
+    # e.g.
+    # curl http://k2.qe.suse.de:8080/size\?product\=fts.jeos\&asset\=fts.SLES15-SP5-Minimal-VM.x86_64-kvm-and-xen-Build3.32.qcow2\&value\=eq.258.1875
```
Comment (jlausuch) [inline lib/db_utils.pm:213]: leftover?

**Review by jlausuch — COMMENTED** [review]
Looks promising! 
Maybe we could discuss if we move the "record exists?" in the push method, and leave the `check_postgres_db` method simpler, just making sure that the DB is reachable?

**Review by jlausuch — APPROVED** [review]

**Pull request merged**